tests/common/mock_environment.py

```diff
@@ -0,0 +1,149 @@
+# -*- coding: utf-8 -*-
+# Copyright 2020 Microsoft Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Requires Python 2.6+ and Openssl 1.0+
+#
+import os
+import re
+import shutil
+import subprocess
+
+from azurelinuxagent.common.utils import fileutil
+from tests.tools import patch, patch_builtin
+
+
+class MockEnvironment:
```

Comment:
As new functionality was added, the mocking logic was getting too long to be defined in a single function so I decided to make this a proper class instead.